src/processor/operator/persistent/reader/parquet/parquet_reader.cpp

```diff
@@ -601,9 +602,11 @@ static common::offset_t tableFunc(TableFuncInput& input, TableFuncOutput& output
     do {
         parquetScanLocalState->reader->scan(*parquetScanLocalState->state, outputChunk);
         if (outputChunk.state->selVector->selectedSize > 0) {
+            parquetScanSharedState->numRowsRead += outputChunk.state->selVector->selectedSize;
```

Comment:
Ditto I think we can use `blockIdx` and `getNumRowsGroups()` in `parquetScanSharedState`. Also try to make the change inside `parquetScanSharedState`.